### DISCUSSION TITLE
Could I apply this package to the time-series raw data?

### DISCUSSION THREAD

**Discussion by ijunglee** [analyticalmindsltd/smote_variants#55] [General]
Hi, I am doing a project which requires to directly input the time-series sensor data, such as acceleration and angular velocity, to the regression-based deep learning model for predicting a score of movement for each subject.
However, I noticed that there are quite few subjects with a certain range of score, and the accuracy of the model dropped when the score of the subject for testing is in this range.
I have read the documentation of SMOTE and it seems that SMOTE-based algorithm are mainly used for augmenting the features, not time-series raw data.
Is that possible to directly apply the SMOTE-based algorithm to the time-series raw data?
Thank you so much in advance!

**Comment by gykovacs** [comment]
Hi @ijunglee ,

well, yes and no. Generally, all SMOTE techniques are focusing on generating samples in a D dimensional feature space. So in this sense, it is not very well suited for time series, which might vary in length (do they?). So if the time series have the same length, SMOTE techniques might work. Depending on how reasonable interpolation between the time series is.

On the flipside, if you work on time series by windowing, that is, extracting numerous windows of the same length, then that dataset is definitely suitable for SMOTE and its variants for oversampling.

Finally, if you have some neural architecture which operates on time series of varying lengths, but at some certain point of the neural architecture, all time series get to the same dimensionality (e.g. using some autoencoder), then at that point, those D dimensional representations are suitable for SMOTE based oversampling.

So long story short, it is not impossible to apply SMOTE to time series, but it depends on how the time series is intended to be processed.

If you are interested to discover applications of SMOTE on time series, we can get into more details, I'm also interested. Also, I have seen some paper titles on "oversampling time series", it might be worth checking in Google, I did not implement any of those methods (didn't even really read the papers), but if the problem of oversampling time series can be transformed into a representation when SMOTE variations are applicable, I'm more than happy to aid the adaptation of the package to time series.

I hope this helped.
